### PR TITLE
chore: Change pending-triage label to pending-maintainer-response

### PR DESCRIPTION
**Pull request by jordanvn**
#### Description of changes

* Newly opened / reopened / transferred issues will now be labelled as "pending maintainer response" rather than "pending triage"
  * This is to bring Amplify UI into parity with other Amplify libraries
* Closed issues will have any "pending triage", "pending maintainer response", and "pending community response" labels removed
* When a community member comments, issues will be labelled as "pending maintainer response" and the "pending community response" label will be removed if it was present
* When an owner comments, the "pending maintainer response" label will be removed if it was present

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/label-new-issues.yml
Comment (reesscot) [inline .github/workflows/label-new-issues.yml:36]: I think transferred tickets are triggered by a different workflow, right?

**Review by reesscot — APPROVED** [review]
LGTM, just wondering if there's another workflow we should update for transferred tickets.

File: .github/workflows/label-new-issues.yml
Comment (jordanvn) [reply .github/workflows/label-new-issues.yml:36]: It looks to me like it's in this workflow. That's what the comments suggest, and it runs with
```
on:
  issues:
    types: [opened, reopened, transferred]
```

**Review by timngyn — DISMISSED** [review]

**Comment by jordanvn** [comment]
Expanded the scope of this change, as bringing Amplify UI into parity with other libraries will introduce new circumstances that these new GitHub workflows handle

**Review by esauerbo — APPROVED** [review]
LGTM, just curious is there a way to validate this works as expected before merging?

**Comment by jordanvn** [comment]
> LGTM, just curious is there a way to validate this works as expected before merging?

Aside from linting, I'm not sure if there is a way to validate the functionality of the on-comment and on-close workflows locally. However, these are the exact scripts from another Amplify repo which previously went through this change.

**Review by reesscot — APPROVED** [review]
LGTM

**Pull request merged**
